### DISCUSSION TITLE
Bulk event import

### DISCUSSION THREAD

**Discussion by penalosa** [plausible/analytics#1392] [Anything Else]
Hi! I've got a somewhat strange use case that involves using Plausible to track listeners to an audio stream. I've got a corn job running that checks the current listener count every minute, and I'm basically looking for a way to injest that into Plausible. I have listener IP addresses & user agents, so I was _thinking_ I could just add a page view event per listener per minute, and that hopefully Plausible would correctly dedupe that based on IP address to get the average number of listeners per hour displayed in the UI (with country & device attribution). However, that'd mean making a _lot_ of requests (one per listener per minute), and so I was wondering if there's a way that a bulk event import endpoint could be implemented? (I've absolutely no familiarity with Elixir, or I'd make a PR myself)

**Comment by ukutaht** [comment]
Yeah, it wouldn't be difficult to to add a `/api/event/bulk` endpoint where you could send an array of events instead of a single one.

How many listeners do you have? You could try sending them all as separate HTTP requests as well, it will probably work OK.

You can also try sending them as custom events (like `Listen`) instead of pageviews if you want to separate the listening data out from regular pageviews

**Reply by ukutaht** [reply]
Oh and you're correct, as long as you send the correct IP address and user-agent, Plausible should dedupe the events correctly.

**Reply by penalosa** [reply]
Generally the listener count is pretty low (less than 20 or so), so multiple requests could work, that's a good shout. I'm trying to run this polling within a Cloudflare worker, though, which has a 50 sub request limit (so when we hit 50 listeners—rare, but not unheard of, this could fail). I suppose I could do that for the time being though, thanks! (And thanks for confirming the deduping). In terms of separating the data, I just put it on a separate domain, but a custom event might be a better solution, I'll look into that.

**Reply by ukutaht** [reply]
Yeah, if you click on the custom event, you can still get a graph to see the listener count over time.
